tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Serializers/KnownSerializers/KnownSerializersTests.cs

```diff
@@ -0,0 +1,126 @@
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Linq;
+using MongoDB.Driver.Linq.Linq3Implementation;
+using MongoDB.Driver.Linq.Linq3Implementation.Translators.ExpressionToExecutableQueryTranslators;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Serializers.KnownSerializers
+{
+    public class KnownSerializersTests
+    {
+        private static readonly IMongoClient __client;
+        private static readonly IMongoCollection<C1> __collection;
+        private static readonly IMongoCollection<C2> __collection2;
+        private static readonly IMongoCollection<C3> __collection3;
+        private static readonly IMongoDatabase __database;
+
+        static KnownSerializersTests()
+        {
+            __client = DriverTestConfiguration.Client;
+            __database = __client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+            __collection = __database.GetCollection<C1>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+            __collection2 = __database.GetCollection<C2>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+            BsonClassMap.RegisterClassMap<C3>(cm =>
+            {
+                cm.AutoMap();
+                cm.MapMember(p => p.Es).SetSerializer(new EnumSerializer<E>(BsonType.String));
+            });
+            __collection3 = __database.GetCollection<C3>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+        }
+
+        public enum E { A, B };
+
+        class C1
```

Comment:
Done. Marked all testing-related class definitions as private except for `enum E`, which had to be public because it is used in the `InlineData` attributes on the test methods. Marking the enum as private results in an inconsistent accessibility compiler error.